WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -329,8 +332,24 @@ extension ProductFormViewController: TextViewViewControllerDelegate {
         return NSLocalizedString("Please add a title",
                                  comment: "Product title error notice message, when the title is empty")
     }
+}
 
+// MARK: - Navigation actions handling
+//
+extension ProductFormViewController {
+    override func shouldPopOnBackButton() -> Bool {
+        if product != originalProduct {
```

Comment:
Business logic such as this, including tracking the `originalProduct` and `product`, should be in a `ViewModel`. But we seem to be far off from that so this is okay. 

```swift
return viewModel.hasUnsavedChanges
```

Reply:
yea I think it'd be great to have a separate class/struct managing the view data, like a `ViewModel` so that we can test such behaviors as we implement more complicated logic. It also allows us to reset the view model after updating the product remotely, and `originalProduct` could be immutable. the next time we're adding more logic to this view controller, I'm going to explore extracting the view data logic to a view model!

Reply:
I'm excited to see that! Let me know if I can help with the design in any way. 🙂